src/time.rs

```diff
@@ -0,0 +1,170 @@
+//! Time structures.
+//!
+//! The `time` modules contains structures used to represent both
+//! absolute and relative time.
+
+use core::{ops, fmt};
+
+static MILLIS_PER_SEC: u64 = 1_000;
+
+/// A representation of an absolute time value.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
```

Comment:
`Default` is semantically inapplicable. There are no default instants! Deriving `Default` for something else with an `Instant` in it is a logic error; it should use an `Option<Instant>` instead.